### DISCUSSION TITLE
Senec-Charger startet neu: NPU?

### DISCUSSION THREAD

**Discussion by k-ranseier** [solectrus/solectrus#3647] [Q&A] [answered]
N´abend alleseits. 

Vor zwei Wochen habe ich Solectrus mittels Konfigurator neu erstellt. In diesem Zuge bin ich dann bei einigen containern zurück von :develop auf :latest. Auch benutze ich nun solcast als Sonnenvorhersage.
Gestartet und funktioniert auf Anhieb. Dann noch händisch (wieso eigentlich nicht über den Konfigurator?) um den tibber-collector und senec-charger ergänzt. Neu komponiert, startet auf Anhieb ohne Fehler und macht was es soll. Super!!

Jetzt startet der senec-charger allerdings alle paar Tage neu. Ich habe den NPU-Fehler im Verdacht. Von der Uhrzeit her würde es passen.
Ich frage den Ladestand allerdings auch alle 10 Minuten ab...

**2024-11-07 21:58:56 [I|NPU] Switched off

senec-charger-1  | #325 - 2024-11-07 22:53:43 +0100
senec-charger-1  | bundler: failed to load command: app/main.rb (app/main.rb)**

log von gestern abend:
[charger-log.txt](https://github.com/user-attachments/files/17682871/charger-log.txt)

Wenn der charger dann endlich mal durchgestartet ist läuft es bis zum nächsten container-start ohne Probleme weiter. Nicht schön, aber gibt wohl schlimmeres...
Aber was passiert wenn der charger mitten im "manuellen Not-Laden" durch den NPU-Fehler irritiert wird..?

MfG und schönes WE

**Comment by ledermann** [comment]
Deinem Log lässt sich entnehmen, dass das Problem ein `Net::OpenTimeout` ist.

Grundsätzlich sollte der SENEC-Speicher nicht mit Anfragen überhäuft werden, das kann sich negativ auf dessen Antwortzeiten auswirken und den NPU-Fehler provozieren, siehe hier:
https://github.com/orgs/solectrus/discussions/3486

Hast du vielleicht noch andere Tools im Einsatz, die den SENEC-Speicher regelmäßig kontaktieren? Es könnte sinnvoll sein, die Abfragehäufigkeit zu reduzieren.

> Aber was passiert wenn der charger mitten im "manuellen Not-Laden" durch den NPU-Fehler irritiert wird..?

Wenn die Notbeladung ausgelöst wird, lädt der Speicher voll und hält die Ladung. Sie muss dann explizit freigegeben werden. Der SENEC-Charger macht das auch. Wird der Container jedoch zwischendurch beendet, wird dieses Kommando nicht gegeben. Wenn der Container aber irgendwann dann doch wieder gestartet wird, erfolgt auch die Freigabe. Insofern also kein Problem.

Dass der SENEC-Charger bei einem Timeout hart aussteigt, ist natürlich nicht so schön. Ich habe solectrus/senec-charger#183 angelegt.

**Reply by k-ranseier** [reply, marked as answer]
Hallo Georg

Auf dem "Solectrus-Raspi" läuft parallel lediglich evcc. 

Ich habe allerdings noch HomeAssistant auf einem NUC laufen. Dort wurde auch der SENEC abgefragt.
Ich habe das AddOn mal deinstalliert.

Schaunmermal...
